### PR TITLE
refactor: use better assertj assertions.

### PR DESCRIPTION
**Pull request by jplaisted**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-builders/src/test/java/com/linkedin/metadata/builders/search/DatasetIndexBuilderTest.java [hunk 1]
```diff
@@ -52,9 +52,7 @@ public void datasetPropertiesSetsDescription() {
     final List<DatasetDocument> actualDocs = _indexBuilder.getDocumentsToUpdate(datasetSnapshot);
 
     // then
-    assertThat(actualDocs.size()).isEqualTo(2);
-    assertThat(actualDocs).contains(expectedDocument1);
-    assertThat(actualDocs).contains(expectedDocument2);
+    assertThat(actualDocs).containsExactly(expectedDocument1, expectedDocument2);
```
Comment (jywadhwani) [inline metadata-builders/src/test/java/com/linkedin/metadata/builders/search/DatasetIndexBuilderTest.java:55]: looks like order matters for `containsExactly` but does it matter to us? Is `containsExactlyInAnyOrder` better?

**Comment by jplaisted** [comment]
AssertJ should not be used as if it is any other assertion library, it isn't. It is way better :)

You should try to use the most specific an concise assertion possible. Let's look at why. Consider these two snippets.

```
    // then
    assertThat(actualDocs.size()).isEqualTo(2); // 1
    assertThat(actualDocs).contains(expectedDocument1); // 2
    assertThat(actualDocs).contains(expectedDocument2); // 3
```

vs

```
    // then
    assertThat(actualDocs).containsExactly(expectedDocument1, expectedDocument2);
```

They do basically the same thing[1]. However, we only need 1 line that is easier to read in the second snippet to accomplish what the first did in 3. Additionally, consider what happens when asserts fail.

1. `Expected 2 but got <actual>` - This looks like we were comparing numbers, and makes no mention of a list. It won't tell us what is missing / what was unexpected.
2. (and 3) `Expected list [<elements>] to contain <actual>, but it did not`. This is a better assert, but imagine that 2 fails. 3 could also be failing, we're not sure. It is a failure in isolation; it just tells us that the element was not in our list, but we really want to express that the entire list contained only these elements.


Now consider if the new snippet fails, for any reason. It will still print the reason, but there could also be more than one. It will also always print the expected contents of the list, as well as the actual contents of the list, making it quite easy to see where the failure was. Imagine that the second element was different. Our assertion could look like:

```
Expecting:
  <[{urn=urn:li:dataset:(urn:li:dataPlatform:foo,bar,PROD), description=baz},
    {urn=urn:li:dataset:(urn:li:dataPlatform:foo,bar,PROD), browsePaths=[/prod/foo/bar], origin=PROD, name=bar, platform=foo}]>
to contain exactly (and in same order):
  <[{urn=urn:li:dataset:(urn:li:dataPlatform:foo,bar,PROD), description=baz},
    {urn=urn:li:dataset:(urn:li:dataPlatform:foo,bar,PROD), browsePaths=[/prod/foo/baz], origin=PROD, name=bar, platform=foo}]>
but some elements were not found:
  <[{urn=urn:li:dataset:(urn:li:dataPlatform:foo,bar,PROD), browsePaths=[/prod/foo/baz], origin=PROD, name=bar, platform=foo}]>
and others were not expected:
  <[{urn=urn:li:dataset:(urn:li:dataPlatform:foo,bar,PROD), browsePaths=[/prod/foo/bar], origin=PROD, name=bar, platform=foo}]>
```

In the first snippet, this example would've failed on the third assert, stating that `Expected list [<elements>] to contain <actual>, but it did not`. This message makes no mention of the bad element.


tl;dr: use the most specific asserts possible in assertj.


---

[1] `containsExactly` expects the same order, where as the original snippet with `contains` does not. You can use `containsExactlyInAnyOrder` if you don't care about order, but it doesn't hurt to test that.

--

More examples, since we're here, but collections have very specific asserts with assertj. Use them. Generally speaking, prefer `assertThat(collection).methodAssertion()` rather than `assertThat(collection.method()).assertion()`.

Example, prefer `assertThat(collection).isEmpty()` over `assertThat(collection.isEmpty()).isTrue()`. Again, you'll get much clearer error messages on failure.

File: metadata-builders/src/test/java/com/linkedin/metadata/builders/search/DatasetIndexBuilderTest.java (hunk 1 above)
Comment (jplaisted) [reply metadata-builders/src/test/java/com/linkedin/metadata/builders/search/DatasetIndexBuilderTest.java:55]: Good question. Let me think that over some more.

Comment (jplaisted) [reply metadata-builders/src/test/java/com/linkedin/metadata/builders/search/DatasetIndexBuilderTest.java:55]: I think since we're exposing a `List` rather than `Collection`, we should probably assert order.
